geode-core/src/main/java/org/apache/geode/internal/cache/AbstractRegionMap.java

```diff
@@ -838,11 +838,11 @@ public boolean initialImagePut(final Object key, final long lastModified, Object
                 if (result) {
                   if (oldIsTombstone) {
                     owner.unscheduleTombstone(oldRe);
-                    if (newValue != Token.TOMBSTONE) {
```

Comment:
How about putting the `lruEntryUpdate(oldRe)` and `lruEntryCreate(oldRe)`calls inside the `else` for the case in which the value for the new entry is not a tombstone as follows:

```
...      
                  } else { //newValue != Token.TOMBSTONE
                    int newSize = owner.calculateRegionEntryValueSize(oldRe);
                    if (!oldIsTombstone) {
                      lruEntryUpdate(oldRe);
                      owner.updateSizeOnPut(key, oldSize, newSize);
                    } else {
                      lruEntryCreate(oldRe);
                      owner.updateSizeOnCreate(key, newSize);
                    }
...
```

Reply:
Thanks for taking a look at these changes. I mainly put this PR up to see if it would pass the unit tests. This was the change that I was concerned about. Event though the tests passed, I'm still not sure its a valid change. I have a request to another team to take a look at these changes before I proceed.

Reply:
I agree with Alberto. I don't see any reason to do call lruEntryUpdate/lruEntryCreate if the next thing we are going to do is call lruEntryDestroy. But it does seem like this code should make sure to call one of lruEntry(Create/Update/Destroy). So I would change it to be:
`                if (result) {
                  if (oldIsTombstone) {
                    owner.unscheduleTombstone(oldRe);
                  }
                  if (newValue == Token.TOMBSTONE) {
                    if (!oldIsDestroyedOrRemoved) {
                      owner.updateSizeOnRemove(key, oldSize);
                    }
                    owner.scheduleTombstone(oldRe, entryVersion);
                    lruEntryDestroy(oldRe);
                  } else {
                    int newSize = owner.calculateRegionEntryValueSize(oldRe);
                    if (!oldIsTombstone) {
                      owner.updateSizeOnPut(key, oldSize, newSize);
                      lruEntryUpdate(oldRe);
                    } else {
                      owner.updateSizeOnCreate(key, newSize);
                      lruEntryCreate(oldRe);
                    }
                    EntryLogger.logInitialImagePut(_getOwnerObject(), key, newValue);
                  }
                }
`

Reply:
Great. Thanks a lot @dschneider-pivotal. I'll make the changes you suggest.